File: lib/python/opentoken/src/test/opentoken/processor/token_constants_test.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from opentoken.processor.token_constants import TokenConstants


class TestTokenConstants:
    """Test cases for TokenConstants."""

    def test_token_constants_exist(self):
        """Test that all token constants exist."""
        assert hasattr(TokenConstants, 'TOKEN')
        assert hasattr(TokenConstants, 'RULE_ID')
        assert hasattr(TokenConstants, 'RECORD_ID')

    def test_token_constants_values(self):
        """Test that token constants have correct values."""
        assert TokenConstants.TOKEN == "Token"
        assert TokenConstants.RULE_ID == "RuleId"
        assert TokenConstants.RECORD_ID == "RecordId"

    def test_token_constants_immutable(self):
        """Test that token constants are not None."""
        assert TokenConstants.TOKEN is not None
        assert TokenConstants.RULE_ID is not None
        assert TokenConstants.RECORD_ID is not None

File: lib/java/opentoken/src/test/java/com/truveta/opentoken/processor/TokenConstantsTest.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.processor;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

import org.junit.jupiter.api.Test;

class TokenConstantsTest {

    @Test
    void testTokenConstantsExist() {
        assertNotNull(TokenConstants.TOKEN);
        assertNotNull(TokenConstants.RULE_ID);
        assertNotNull(TokenConstants.RECORD_ID);
    }

    @Test
    void testTokenConstantsValues() {
        assertEquals("Token", TokenConstants.TOKEN);
        assertEquals("RuleId", TokenConstants.RULE_ID);
        assertEquals("RecordId", TokenConstants.RECORD_ID);
    }
}
